docs/code-style.md

```diff
@@ -0,0 +1,132 @@
+# Code style
+
+## JavaScript
+
+We use [ESlint](https://eslint.org/) and [Prettier](https://prettier.io/) to enforce our JavaScript code style. 
+Running `make fix` after making any changes will fix most things.
+
+There are some styles the linter can't pick up. If you are unsure of anything, [AirBnB's style
+guide](https://github.com/airbnb/javascript) is a good place to look.
+
+## Components
+
+We use [React](https://reactjs.org/) for our components, and [Emotion](https://emotion.sh/) for our CSS in JS.
+
+- Use Emotion's [`css`](https://emotion.sh/docs/css) function to build a class name from styles
+
+> Why? It decouples our CSS declaration from specific components
+
+```js
+// bad
+const MyComponent = styled('div')`
+    color: red;
+`;
+render(
+    <MyComponent />
+);
+
+// bad
+render(
+    <div css={`
+        color: red;
+    `} />
+);
+
+// good
+const myComponent = css`
+    color: red;
+`;
+render(
+    <div className={myComponent} />
+);
+```
+
+- Extract CSS into a variable rather than defining it inline in a component
+
+> Why? It keeps the JSX tidy
+
+```js
+// bad
+render(
+    <div className={css`
+        color: red;
+    `} />
+)
+
+// good
+const myComponent = css`
+    color: red;
+`;
+render(
+    <div className={myComponent} />
+);
+```
+
+- Extract dynamic styles into a function that takes `props`
+
+> Why? It avoids defining styles inline in a component, so keeps the JSX tidy
+
+```js
+// bad
+const MyComponent = ({ fontColor }) =>
+    <div className={css`
+        color: ${fontColor};
+    `} />;
+
+// good
+const myComponent = (fontColor) => css`
+    color: ${fontColor};
+`;
+const MyComponent = ({ fontColor }) =>
```

Comment:
Is it worth calling out that in general, Emotion seems to suggest avoiding dynamic styles for performance reasons - because it wants you to minimise the number of classes it has to generate. This isn't always possible of course, but quite often is e.g. by doing stuff like:

    const pink = css`
       color: pink;
    `

    const red = css`
       color: red;
    `

    const color = (someBoolean) => {
        if (someBoolean) {
            return red;
        } else {
            return pink;
        }
    }

So perhaps we should discuss this as it is a common case where the obvious solution and the 'correct' solution are not the same.

Reply:
> Is it worth calling out that in general, Emotion seems to suggest avoiding dynamic styles for performance reasons - because it wants you to minimise the number of classes it has to generate

I know this may be true of styled-components, can you find any documentation that suggests it is true of Emotion?

Reply:
Hey, so I think my understanding of this was slightly wrong; it does seem like there are costs to dynamic styles, but not for quite the reasons I thought.

There's not really docs on this, but from sleuthing it seems like the css helper *hashes* it's content to generate the class name. So it won't generate a new class if it doesn't have to. So in that sense dynamic styles are fine.

*However*, the hashing process itself is not fast, so doing it lots of times can slow down your app. E.g. see here: https://github.com/emotion-js/emotion/issues/536#issuecomment-357552694.

I don't think this is likely to be a problem anytime soon though so, apart from pathalogical cases (large collections with dynamic styles), we can probably just do regular functions/dynamic behaviour.

Reply:
Cool, thanks for that detective work. Shall we leave this as-is for now?

Reply:
Yep. Your comment https://github.com/guardian/dotcom-rendering/pull/126#discussion_r209999843 here was also helpful! Thanks.

Reply:
Yep thanks!